lib/computer_player.rb

```diff
@@ -1,17 +1,80 @@
+require_relative 'win_checker'
+require_relative 'score_tracker'
+
 class ComputerPlayer < Player
 
-  def initialize(player_num, board)
-    @board = board
+  def initialize(player_num, side_length)
+    @player_num = player_num
+    @next_move = 0
+    @side_length = side_length
+    @comp_token = "O"
+    @other_token = "X"
   end
 
-  def get_next_move
-    available_moves = @board.get_available_moves
+  def get_next_move(board)
+    minimax_score(board.to_string_array, @player_num)
 
-    available_moves[0].to_s
+    @next_move.to_s
   end
 
   def get_token
-    "O"
+    @comp_token
+  end
+
+  private
+
+  def minimax_score(board_string_array, next_player)
+    current_board = Board.new(@side_length)
+    current_board.set_board(board_string_array)
+    score_tracker = ScoreTracker.new(current_board)
```

Comment:
It seems that the ScoreTracker object is created, used once, and then discarded.  This seems unnecessary.  If you pass the board into `get_player_score` instead, you can have one object that does calculates scores instead of thousands objects that calculate scores.

Reply:
Only creating one ScoreTracker object now. The board is being passed to update_scores, rather than get_player_score.